### PR TITLE
Compression updates

### PR DESCRIPTION
**Pull request by rtibbles**
* Adds compression for audio files
* Adds compression for embedded media files in archive file formats
* Adds global configuration options for compression
* Fixes windows tests
* Fixes Ubuntu/Python 3.6 tests

### CODE REVIEW

File: ricecooker/utils/zip.py [hunk 1]
```diff
@@ -47,24 +48,32 @@ def reader(x):
...
-            return inputzip.read(x)
+            def reader(x):
+                return inputzip.read(x)
 
-    else:
-        raise Exception("The `path` must either point to a directory or to a zip file.")
+        except Exception:
+            raise Exception(
+                "The `path` must either point to a directory or to a zip archive."
+            )
```
Comment (nucleogenesis) [inline ricecooker/utils/zip.py:63]: I feel like I've been yelled at by a linter for trying to except a base `Exception` and had to make my own subclass of it to pass - is this due to lack of parity between Ricecooker and other repos on linting or am I misremembering?

File: tests/media_utils/test_audio.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        os.remove(filename)
+    except FileNotFoundError:
+        pass
+    assert not os.path.exists(filename)
+
+
+class TempFile(object):
+    """
+    tempfile.NamedTemporaryFile deletes the file as soon as the filehandle is closed.
+    This is OK on unix but on Windows the file can't be used by other commands
```
Comment (nucleogenesis) [inline tests/media_utils/test_audio.py:84]: Is this problem resolved primarily by passing `delete=True` to the `NamedTemporaryFile` and then the context manager business just ensures no matter what w/ the `atexit` registration that the files are always deleted?

**Review by nucleogenesis — APPROVED** [review]
Looks good just a few questions overall.

I wonder if that context manager `TempFile` would be helpful to use in any of the production code where we use `NamedTemporaryFile`? Maybe there is some risk of problems on Windows that this being exclusively used in the test suite obfuscates?

I suppose that if there haven't been reports of such issues from our Windows user that it probably isn't the case, but next time I mess with any kind of temporary files I'll probably come borrow that context manager class to simplify and be sure of the clean up.

File: ricecooker/utils/zip.py (hunk 1 above)
Comment (rtibbles) [reply ricecooker/utils/zip.py:63]: The only thing I can think of that a linter might shout at you for is a bare `except` clause without an `Exception` class or subclass specified.

File: tests/media_utils/test_audio.py (hunk 2 above)
Comment (rtibbles) [reply tests/media_utils/test_audio.py:84]: So, it's to do with accessing the file from a subprocess (using e.g. ffmpeg), this basically lets us treat it like a regular context manager for a temporary file, but still have the deletion.

Note that I copied this from the `test_video.py` and did not come up with this myself, I think...

**Pull request merged**